cdm/core/src/main/java/ucar/nc2/ft/point/writer/CFPointWriter.java

```diff
@@ -147,22 +150,21 @@ private static int writeStationFeatureCollection(FeatureDatasetPoint dataset, St
 
       cfWriter.setExtraVariables(fc.getExtraVariables());
 
-      ucar.nc2.ft.PointFeatureCollection pfc = fc.flatten(null, null, null); // all data, but no need to sort by station
+      cfWriter.writeHeader(fc);
 
       int count = 0;
-      for (PointFeature pf : pfc) {
-        StationPointFeature spf = (StationPointFeature) pf;
-        if (count == 0)
-          cfWriter.writeHeader(fc.getStationFeatures(), spf);
+      for (PointFeatureCollection pfc : fc) {
+        for (PointFeature pf : pfc) {
+          StationPointFeature spf = (StationPointFeature) pf;
 
-        cfWriter.writeRecord(spf.getStation(), pf, pf.getFeatureData());
-        count++;
-        if (debug && count % 100 == 0)
-          System.out.printf("%d ", count);
-        if (debug && count % 1000 == 0)
-          System.out.printf("%n ");
+          cfWriter.writeRecord(spf.getStation(), pf, pf.getFeatureData());
+          count++;
+          if (debug && count % 100 == 0)
```

Comment:
I know you didn't add this but can we get rid or this or change it to logger.debug?